File: 송하연/10월 2주차/순위.java
import java.util.*;

class Solution {
    int answer = 0;
    
    public int solution(int n, int[][] results) {

        List<List<Integer>> winGraph = new ArrayList<>();
        List<List<Integer>> loseGraph = new ArrayList<>();
        
        for(int i=0; i<n; i++){
            winGraph.add(new ArrayList<>());
            loseGraph.add(new ArrayList<>());
        }
        
        for(int[] group:results){
            int win = group[0];
            int lose = group[1];
            winGraph.get(win-1).add(lose-1);
            loseGraph.get(lose-1).add(win-1);
        }
        
        //이긴 사람+진 사람 수 = n-1
        for(int i=0; i<n; i++){
            boolean[] visited = new boolean[n];
            visited[i] = true;
            int winCnt = dfs(winGraph,i,visited);
            
            Arrays.fill(visited,false);
            visited[i] = true;
            int loseCnt = dfs(loseGraph,i,visited);
            
            if(winCnt+loseCnt==n-1){
                answer++;
            }
        }
        
        return answer;
    }
    
    int dfs(List<List<Integer>> graph, int node, boolean[] visited){
        int cnt=0;
        for(int next:graph.get(node)){
            if(!visited[next]){
                visited[next] = true;
                cnt+= 1+dfs(graph,next,visited);
            }
        }
        return cnt;
    }
}

File: 한승우/10월 2주차/순위.cpp
#include <string>
#include <vector>
#include <queue>

using namespace std;

//a > b이고 b > c이면 a > c이다.

bool result[101][101];

int solution(int n, vector<vector<int>> results) {
    int answer = 0;
    
    for(auto& r: results){
        result[r[0]][r[1]] = true;
    }
    
    for(int k= 1; k <= n; k++){
        for(int a = 1; a <= n; a++){
            for(int b = 1; b <= n; b++){
                if(result[a][k] && result[k][b]) {
                    result[a][b] = true;
                }
            }
        }
    }
    
    for(int i = 1; i <= n; i++){
        int cnt = 0;
        
        for(int j = 1; j <= n; j++){
            if(result[i][j] || result[j][i]) {
                cnt++;
            }
        }
        
        if(cnt == n - 1) answer++;
    }
    
    return answer;
}